In [1]:
import my_module as mm
device = mm.torch.device('cuda' if mm.torch.cuda.is_available() else 'cpu')
print(device)

cuda


In [2]:
categories = [
    "anger",
    "surprise",
    "disgust",
    "fear",
    "neutral",
    "happiness",
    "sadness",
    "contempt"
]

# categories = [
#     "anger",
#     "surprise",
#     "disgust",
#     "fear",
#     "neutral",
#     "happiness",
#     "sad",
# ]

X, y = mm.read_pic(mm.pd.read_csv('data/legend.csv'), categories, limit = 1000)
# X, y = mm.read_pic(mm.pd.read_csv('data/500_picts_satz.csv'), categories)
seed = 1234
Xtr, Xva, ytr, yva = mm.train_test_split(X, y, test_size=0.1, random_state=seed, shuffle=True)
trainloader, testloader = mm.get_loader(Xtr, Xva, ytr, yva, batch_size = 1)
print("Done")

Done


In [3]:
convs = [8, 16, 32, 64, 128]
fcs = [512, 64]
net = mm.Net2(1, 350, 350, 8, convs, fcs)

In [4]:
mm.torch.cuda.empty_cache()
count = 0
DIR_PATH = './model/'
mm.os.makedirs(DIR_PATH, exist_ok = True)
if mm.torch.cuda.is_available():
    mm.torch.cuda.empty_cache()
    net.cuda()
net = mm.train_process2(device, net, trainloader, LR = 0.1, EPOCH = 2, PRINT = 100)
count += 1
PATH = DIR_PATH + "model"".pth"
mm.torch.save(net.state_dict(), PATH)
print(PATH, "done.")

CrossEntropyLoss()


OutOfMemoryError: CUDA out of memory. Tried to allocate 1.98 GiB. GPU 0 has a total capacty of 15.73 GiB of which 716.50 MiB is free. Process 588112 has 8.82 GiB memory in use. Process 663806 has 6.20 GiB memory in use. Of the allocated memory 5.96 GiB is allocated by PyTorch, and 26.23 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

In [ ]:
print(f'Training error: {1 - mm.accuracy(device, net, trainloader)}')
print(f'Testing error: {1 - mm.accuracy(device, net, testloader)}')

CUDA out of memory with higher fliter:
OutOfMemoryError: CUDA out of memory. Tried to allocate 3.65 GiB. GPU 0 has a total capacty of 15.73 GiB of which 200.50 MiB is free. Process 377677 has 2.16 GiB memory in use. Process 442884 has 13.36 GiB memory in use. Of the allocated memory 12.82 GiB is allocated by PyTorch, and 327.44 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

The third 